[EP ABI] Utility to serialize OrtGraph to GraphProto

adrianlizarraga opened the pull request:
### Description
- Provides utility functions that serialize an `OrtGraph` to a `GraphProto` or `ModelProto`.
  - Header-only file that can be copied to a project that builds with ORT and ONNX.
  - Available in [include/onnxruntime/core/providers/utils/ort_graph_to_proto.h](https://github.com/microsoft/onnxruntime/blob/adrianl/ep-abi-ort-graph-to-onnx-protobuf/include/onnxruntime/core/providers/utils/ort_graph_to_proto.h)
- Updates the `Node_GetSubgraphs` API function to also return the attribute names associated with each subgraph. This is required to determine which subgraph corresponds to a given attribute.
- Adds `Graph_GetNumOperatorSets` and `Graph_GetOperatorSets` API functions to get the opset version for each domain.



### Motivation and Context
Provide a utility to facilitate porting of existing execution providers to the new EP ABI. The utilities introduced by this PR convert an `OrtGraph` into an ONNX protobuf representation, which some existing EPs currently convert to their internal representation. Ideally, we would prefer a more direct conversion from a `OrtGraph` to the EP's internal representation, but this is a large effort. These utilities enable an incremental transition.

jywu-msft commented:
Thanks! we should discuss where the utility functions should live. there are several options.

adrianlizarraga commented:
@chilo-ms @jywu-msft , I made this a header-only file that can be copied into a project that builds with ORT + ONNX. I moved it into `include/onnxruntime/core/providers/utils/ort_graph_to_proto.h` and added documentation comments describing how to include the file.

Please let me know if you have any feedback. Thanks.

adrianlizarraga commented:
Closing and re-opening because some checks are not running.
